components/automate-ui/src/app/modules/infra-proxy/create-chef-server-modal/create-chef-server-modal.component.spec.ts

```diff
@@ -35,18 +40,164 @@ describe('CreateChefServerModalComponent', () => {
+
+        errors = createForm.controls['id'].errors || {};
+
+        expect(createForm.valid).toBeFalsy();
+        expect(errors['required']).toBeTruthy();
+      });
+
+      it('when fqdn is missing', () => {
+        createForm.controls['name'].setValue('test');
+        createForm.controls['id'].setValue('test');
+        createForm.controls['ip_address'].setValue('1.2.3.4');
+
+        errors = createForm.controls['fqdn'].errors || {};
+
+        expect(createForm.valid).toBeFalsy();
+        expect(errors['required']).toBeTruthy();
+      });
+
+      it('when ip_address is missing', () => {
+        createForm.controls['name'].setValue('test');
+        createForm.controls['id'].setValue('test');
+        createForm.controls['fqdn'].setValue('test.net');
+
+        errors = createForm.controls['ip_address'].errors || {};
+
+        expect(createForm.valid).toBeFalsy();
+        expect(errors['required']).toBeTruthy();
+      });
+
+      it('when the ip_address in invalid', () => {
+        createForm.controls['name'].setValue('test');
+        createForm.controls['id'].setValue('test');
+        createForm.controls['fqdn'].setValue('chef.internal');
+
+        createForm.controls['ip_address'].setValue('1.2234.3.4');
+        errors = createForm.controls['ip_address'].errors || {};
+
+        expect(createForm.valid).toBeFalsy();
+        expect(errors['pattern']).toBeTruthy();
+      });
+
+      // Many testing ideas attributed to https://mathiasbynens.be/demo/url-regex
+
+      using([
+        ['is using something other than http or https', 'httpld://www.chef.io'],
+        ['contains two periods', 'chef..internal'],
+        ['there is no TLD suffix', 'http://foo.com.'],
+        ['contains hypens in the TLD', 'chef.this-will-not'],
+        ['has a TLD that is longer than 25 characters', 'chef.thisisareallylongtldandwontwork'],
+        ['has a TLD that is shorter than 2 characters', 'chef.i'],
+        ['has numbers in the TLD', 'chef.017'],
+        ['has a port number that is too high', 'https://chef.io:987274892'],
+        ['has a colon but no port number', 'https://chef.io:'],
+        ['has a letter in the port', 'https://chef.io:123a'],
+        ['has no domain', 'http://'],
+        ['has no secure domain', 'https://'],
+        ['domain is dots', 'https://..'],
+        ['domain is hash', 'http://#'],
+        ['domain has a space', 'http:// shouldfail.net'],
+        ['contains all numbers', 'http://10.1.1.0']
+      ], function (description: string, input: string) {
+        it(('when the fqdn ' + description), () => {
+          createForm.controls['name'].setValue('test');
+          createForm.controls['id'].setValue('test');
+          createForm.controls['ip_address'].setValue('1.2.3.4');
+
+          createForm.controls['fqdn'].setValue(input);
+          errors = createForm.controls['fqdn'].errors || {};
+
+          expect(createForm.valid).toBeFalsy();
+          expect(errors['pattern']).toBeTruthy();
```

Comment:
While I agree we could have hundreds of tests here, I feel like that is a bit overkill for us, and that these dozen or so are a good cross section.

same feelings for Valid tests below.